### PR TITLE
[Fix] Fix typos, init funcs, cacheable method of part3 of data transforms

### PR DESCRIPTION
**Pull request by plyfager**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/transforms/processing.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def __repr__(self):
 @TRANSFORMS.register_module()
 class CenterCrop(BaseTransform):
     """Crop the center of the image, segmentation masks, bounding boxes and key
-    points. If the crop area exceeds the original image and ``pad_mode`` is not
+    points. If the crop area exceeds the original image and ``do_pad`` is not
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:419]: is not None?

File: mmcv/transforms/processing.py [hunk 2]
```diff
@@ -444,13 +444,8 @@ class CenterCrop(BaseTransform):
...
             width and height are equal to this integer.
-        pad_val (Union[Number, Dict[str, Number]]): A dict for
-            padding value. To specify how to set this argument, please see
-            the docstring of class ``Pad``. Defaults to
-            ``dict(img=0, seg=255)``.
-        pad_mode (str, optional): Type of padding. Should be: 'constant',
-            'edge', 'reflect' or 'symmetric'. For details, please see the
-            docstring of class ``Pad``. Defaults to 'constant'.
+        do_pad (bool): Whether to pad the image if it's smaller than the
+            ``crop_size``. Defaults to False.
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:448]: The type of `pad_cfg` is dict instead of str

File: mmcv/transforms/processing.py [hunk 3]
```diff
@@ -884,24 +881,13 @@ class RandomMultiscaleResize(BaseTransform):
...
-        keep_ratio (bool): Whether to keep the aspect ratio when
-            resizing the image. Defaults to False.
-        clip_object_border (bool): Whether clip the objects outside
-            the border of the image.  Defaults to True.
-        backend (str): Image resize backend, choices are 'cv2' and
-            'pillow'. These two backends generates slightly different results.
-            Defaults to 'cv2'.
-        interpolation (str): The mode of interpolation, support
-            "bilinear", "bicubic", "nearest". Defaults to "bilinear".
+        resize_cfg (dict): Base config for resizing. Defaults to
```
Comment (Harold-lkk) [inline mmcv/transforms/processing.py:884]: link to `mmcv.Resize`

File: mmcv/transforms/processing.py [hunk 4]
```diff
@@ -444,14 +444,9 @@ class CenterCrop(BaseTransform):
...
-            padding value. To specify how to set this argument, please see
-            the docstring of class ``Pad``. Defaults to
-            ``dict(img=0, seg=255)``.
-        pad_mode (str, optional): Type of padding. Should be: 'constant',
-            'edge', 'reflect' or 'symmetric'. For details, please see the
-            docstring of class ``Pad``. Defaults to 'constant'.
-        pad_cfg (str): Base config for padding. Defaults to
+        do_pad (bool): Whether to pad the image if it's smaller than the
+            ``crop_size``. Defaults to False.
+        pad_cfg (dict): Base config for padding. Defaults to
```
Comment (Harold-lkk) [inline mmcv/transforms/processing.py:449]: Link to `mmcv.Pad`

File: mmcv/transforms/processing.py [hunk 5]
```diff
@@ -444,14 +444,9 @@ class CenterCrop(BaseTransform):
...
             width and height are equal to this integer.
-        pad_val (Union[Number, Dict[str, Number]]): A dict for
-            padding value. To specify how to set this argument, please see
-            the docstring of class ``Pad``. Defaults to
-            ``dict(img=0, seg=255)``.
-        pad_mode (str, optional): Type of padding. Should be: 'constant',
-            'edge', 'reflect' or 'symmetric'. For details, please see the
-            docstring of class ``Pad``. Defaults to 'constant'.
-        pad_cfg (str): Base config for padding. Defaults to
+        do_pad (bool): Whether to pad the image if it's smaller than the
```
Comment (ZwwWayne) [inline mmcv/transforms/processing.py:447]: do_pad -> pad

Comment (plyfager) [reply mmcv/transforms/processing.py:448]: I will use ``auto_pad`` instead.

File: mmcv/transforms/processing.py [hunk 6]
```diff
@@ -444,30 +443,25 @@ class CenterCrop(BaseTransform):
...
-            ``dict(img=0, seg=255)``.
-        pad_mode (str, optional): Type of padding. Should be: 'constant',
-            'edge', 'reflect' or 'symmetric'. For details, please see the
-            docstring of class ``Pad``. Defaults to 'constant'.
-        pad_cfg (str): Base config for padding. Defaults to
-            ``dict(type='Pad')``.
+        auto_pad (bool): Whether to pad the image if it's smaller than the
+            ``crop_size``. Defaults to False.
+        pad_cfg (dict): Base config for padding. Refer to ``mmcv.Pad`` for
+            detail. Defaults to ``dict(type='Pad')``.
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:449]: The default value of `pad_cfg` is not only type.

File: mmcv/transforms/processing.py [hunk 7]
```diff
@@ -910,29 +902,21 @@ def __init__(
...
-        self.clip_object_border = clip_object_border
-        self.backend = backend
-        self.interpolation = interpolation
-
         self.resize_cfg = resize_cfg
 
-    @staticmethod
-    def random_select(scales: List[Tuple]) -> Tuple[Number, int]:
+    @cacheable_method
+    def _random_select(self) -> Tuple[Number, int]:
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:927]: Does the type of `scale` is a number?

File: mmcv/transforms/processing.py [hunk 8]
```diff
@@ -884,24 +886,14 @@ class RandomMultiscaleResize(BaseTransform):
...
-        interpolation (str): The mode of interpolation, support
-            "bilinear", "bicubic", "nearest". Defaults to "bilinear".
+        resize_cfg (dict): Base config for resizing. Refer to
+            ``mmcv.Resize`` for detail. Defaults to
+            ``dict(type='Resize')``.
     """
 
     def __init__(
         self,
         scales: Union[list, Tuple],
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:915]: `Sequence[Union[int, Tuple]]`

File: mmcv/transforms/processing.py [hunk 9]
```diff
@@ -929,29 +921,21 @@ def __init__(
...
-        self.backend = backend
-        self.interpolation = interpolation
-
         self.resize_cfg = resize_cfg
 
-    @staticmethod
-    def random_select(scales: List[Tuple]) -> Tuple[Number, int]:
+    @cacheable_method
+    def _random_select(self) -> Tuple[Number, int]:
         """Randomly select an img_scale from given candidates.
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:928]: `img_scale` -> `scale`

**Review by mzr1996 — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**